editoast/src/infra.rs

```diff
@@ -208,6 +208,29 @@ impl Infra {
         }
     }
 
+    pub fn clone(infra_id: i64, conn: &mut PgConnection) -> Result<Infra, Box<dyn ApiError>> {
+        let infra_to_clone = Infra::retrieve(conn, infra_id)?;
```

Comment:
We don't need to fetch the cloned infra. Everything can be perform in a single query.